distribution/sdk/sample-mpegts-stream-generator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,277 @@
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecuteResultHandler;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.exec.Executor;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.SystemUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.DatagramPacket;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+
+/**
+ * This client is used for testing/development to transmit an MPEG-TS file as a stream of UDP
+ * packets.
+ */
+public class MpegTsUdpClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MpegTsUdpClient.class);
+
+    private static final int PACKET_SIZE = 188;
+
+    private static final String DEFAULT_IP = "127.0.0.1";
+
+    private static final int DEFAULT_PORT = 50000;
+
+    private static final String SUPPRESS_PRINTING_BANNER_FLAG = "-hide_banner";
+
+    private static final String INPUT_FILE_FLAG = "-i";
+
+    private static final boolean DONT_HANDLE_QUOTING = false;
+
+    public static void main(String[] args) {
+
+        String[] arguments = args[0].split(",");
+
+        if (arguments.length < 1) {
+            LOGGER.error("Unable to start stream : no arguments specified.");
+            return;
+        }
+
+        String videoFilePath = arguments[0];
+        if (StringUtils.isBlank(videoFilePath)) {
+            LOGGER.error("Unable to start stream : no video file path specified.");
+            return;
+        }
+
+        String ip;
+        int port;
+
+        if (arguments.length == 1) {
+            ip = DEFAULT_IP;
+            port = DEFAULT_PORT;
+        } else if (arguments.length == 2) {
+            ip = arguments[1];
+            port = DEFAULT_PORT;
+        } else {
+            ip = arguments[1];
+            try {
+                port = Integer.parseInt(arguments[2]);
+            } catch (NumberFormatException e) {
+                LOGGER.error("Unable to parse specified port : {}.  Using Default : {}",
+                        arguments[2],
+                        DEFAULT_PORT);
+                port = DEFAULT_PORT;
+            }
+        }
+
+        if (StringUtils.isBlank(ip)) {
+            LOGGER.error("Unable to start stream : no ip address sspecified.");
```

Comment:
Recommend pulling input error logging out to a helper method, with more robust usage description.